### PR TITLE
TextDiff: Decouple TextDiff component from post revisions

### PR DESCRIPTION
**Pull request by spen**
### Summary
This PR is an effort towards decoupling some components that have been built during the development of the post revisions work, making them reusable in other areas of the project, easier to unit test and easier to document.

Specifically - here we're moving out the component that shows additions and removals to text/content in to a new component: `TextDiff`.

### Testing
There should be no deviations from the original functionality in post revisions.
To test this:
- Go to a the post editor
- Make some revisions if the post doesn't already have a few
- Click on the 'history' button in the editors header
- You should see removals highlighted in a faint red, additions in a faint blue exactly as before
- Also pay attention to the line splitting, the title shouldn't break changes on to new lines but the changes in the content should.

Also, take a look at the DevDocs entry - does this look reasonable? Is it worth showing multiple examples? 
Go to https://calypso.live/devdocs and search TextDiff in 'UI Components'.

### TODO:
- [x] Add readme for `TextDiff`
- [x] Add `TextDiff` to DevDocs
- [ ] Potentially rename - what do you think?

### CODE REVIEW

File: client/components/text-diff/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		</h2>
+	);
+}
+```
+
+## Props
+
+The following props are available to customize the accordion:
+
+- `changes`: An array of objects representing a content change, of shape:
```
Comment (jblz) [inline client/components/text-diff/README.md]: What do you think about renaming this to `operations`? `copy` isn't so much a change.

File: client/components/text-diff/style.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.text-diff__additions {
+	background: lighten($blue-wordpress, 58%);
+	border-bottom: 2px solid $blue-wordpress;
+}
+
+
+.text-diff__context {
+	// TODO: Should there be specific styles for this class? Should we just remove it altogther?
```
Comment (jblz) [inline client/components/text-diff/style.scss]: The context class could be useful to apply styles (e.g. lightening the text color, collapsing & hiding..) to "copy" ops.

Maybe update this comment w/ some work that could happen there.

File: client/components/text-diff/README.md (hunk 1 above)
Comment (spen) [reply client/components/text-diff/README.md]: Sounds good to me! I think it'd mean we need a bit of a re-work of the would-be `operation.op` property. I was thinking `type` (as in `changes.type`) would be nice anyway, even more fitting with `operations.type`.

File: client/components/text-diff/style.scss (hunk 2 above)
Comment (spen) [reply client/components/text-diff/style.scss]: 👍 :)

File: client/components/text-diff/README.md (hunk 1 above)
Comment (jblz) [reply client/components/text-diff/README.md]: Good thought.  `operations.type` makes good semantic sense to me.

File: client/components/text-diff/style.scss (hunk 2 above)
Comment (spen) [reply client/components/text-diff/style.scss]: I've rebased the comment (and empty ref) out for now - I hadn't realised that context is really just 'anything not changed in this diff' until last night, so I now think it makes sense not to style it all, at least for now :)

File: client/components/text-diff/README.md (hunk 1 above)
Comment (spen) [reply client/components/text-diff/README.md]: I've updated all instances of `changes` to `operations` but left `op` as-is. It's not quite ideal in terms of semantics but avoids having to amend the data returned from the API.

**Comment by jblz** [comment]
This is working well & code looks good.  I added a blank line at the end of the new style file.

**Comment by spen** [comment]
Awesome - thanks @jblz!

**Pull request merged**